addons/l10n_es_edi_tbai/views/report_invoice.xml

```diff
@@ -2,7 +2,7 @@
 <odoo>
     <template id="l10n_es_tbai_external_layout_standard" inherit_id="account.report_invoice_document">
         <xpath expr="//div[@id='qrcode']" position="after">
-            <div name="l10n_es_tbai_qrcode" t-if="o.l10n_es_tbai_is_required" style="page-break-inside: avoid">
```

Comment:
_is_in_chain might be enought.  

What we could do as well is to add L170-171 from _get_l10n_es_tbai_qr to _get_l10n_es_tbai_id instead. 

That way it is also fixed for those that do not update the module and it is consistent. 

@pivi-odoo Let me know what you think.

Reply:
So I've tested your proposition with a python only solution, sadly, it prints a visible QR code. So a change in the check in the xml is still necessary, there is no way around it :S 

I've still added the condition to `_get_l10n_es_tbai_id` as an additional assertion, it's better to have an `''` instead of a stacktrace if someone calls the method without the signature. (For now in standard `_get_l10n_es_tbai_id` is called only in this template, and when computing the qr code, which never happens if the invoice is not in the chain, so for now the condition is not really *needed*, but a *nice to have*).

I've also updated the condition in the xml as you've suggested, it's true it was redundant/overlapping.

Cheers.